File: backend/internal/models/resumo_pecuario.go
package models

import "time"

type PartoPrevistoResumo struct {
	AnimalID          int64      `json:"animal_id"`
	Identificacao     string     `json:"identificacao"`
	GestacaoID        int64      `json:"gestacao_id"`
	DataPrevistaParto *time.Time `json:"data_prevista_parto,omitempty"`
}

type ResumoPecuario struct {
	PrenhesTotal           int                   `json:"prenhes_total"`
	RestricoesAtivasTotal int                   `json:"restricoes_ativas_total"`
	ProducaoHojeLitros    float64               `json:"producao_hoje_litros"`
	ProducaoSemanaLitros  float64               `json:"producao_semana_litros"`
	PartosPrevistos       []PartoPrevistoResumo `json:"partos_previstos"`
}

File: frontend/src/services/resumoPecuario.ts
import api, { type ApiResponse } from "./api";

export type PartoPrevistoResumo = {
  animal_id: number;
  identificacao: string;
  gestacao_id: number;
  data_prevista_parto?: string | null;
};

export type ResumoPecuario = {
  prenhes_total: number;
  restricoes_ativas_total: number;
  producao_hoje_litros: number;
  producao_semana_litros: number;
  partos_proximos_7d_total: number;
  lactacao_ativa_total: number;
  partos_previstos: PartoPrevistoResumo[];
};

export async function getResumoPecuario(
  fazendaId: number,
  diasParto = 30
): Promise<ResumoPecuario | null> {
  const { data } = await api.get<ApiResponse<ResumoPecuario>>(
    `/api/v1/fazendas/${fazendaId}/resumo-pecuario`,
    { params: { dias_parto: diasParto } }
  );
  return data.data ?? null;
}
